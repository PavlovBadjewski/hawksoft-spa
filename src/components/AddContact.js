import React, { useRef } from 'react';
import { FaWindowClose } from "react-icons/fa";

const AddContact = ({
  hideForm,
  createContact
}) => {

  let firstNameRef = useRef();
  let lastNameRef = useRef();
  let emailRef = useRef();
  let addressRef = useRef();

  const create = () => {
    createContact({
      firstName: firstNameRef.current.value,
      lastName: lastNameRef.current.value,
      email: emailRef.current.value,
      address: addressRef.current.value
    });

    reset();
  }

  const hide = () => {
    hideForm(false);
    reset();
  }

  const reset = () => {
    firstNameRef.current.value = '';
    lastNameRef.current.value = '';
    emailRef.current.value = '';
    addressRef.current.value = '';
  }

  return(
    <div>
      <div className="close-add-icon">
        <FaWindowClose onClick={() => hide()} />
      </div>
      <div className="contact-fields">
        <div>
          <label>First Name:</label>
          <input type="text" ref={firstNameRef} />
        </div>
        <div>
          <label>Last Name:</label>
          <input type="text" ref={lastNameRef} />
        </div>
        <div>
          <label>Email:</label>
          <input type="text" ref={emailRef} />
        </div>
        <div>
          <label>Address:</label>
          <input type="text" ref={addressRef} />
        </div>
      </div>
      <button className="add-contact-button" onClick={() => create()}>Add</button>
    </div>
  );
}

export default AddContact;