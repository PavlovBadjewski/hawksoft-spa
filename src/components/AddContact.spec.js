import React from 'react';
import { shallow, mount } from 'enzyme';
import AddContact from './AddContact';

const defaultProps = {
  hideForm: jest.fn(),
  createContact: jest.fn()
};

describe('AddContact tests', () => {
  afterEach(() => {
    defaultProps.hideForm.mockClear(),
    defaultProps.createContact.mockClear()
  });

  it('it should render correctly', () => {
    // ARRANGE && ACT
    const tree = shallow(<AddContact {...defaultProps} />);
  
    // ASSERT
    expect(tree).toMatchSnapshot();
  });
  
  it('when window close is clicked hideForm is called', () => {
    // ARRANGE && ACT
    const tree = mount(<AddContact {...defaultProps} />);
    const windowCloseButton = tree.find("FaWindowClose").first();
    windowCloseButton.simulate('click');

    //ASSERT 
    expect(defaultProps.hideForm).toHaveBeenCalled();
    expect(defaultProps.createContact).not.toHaveBeenCalled();
  });
});