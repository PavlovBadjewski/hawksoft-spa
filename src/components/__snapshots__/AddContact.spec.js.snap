// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContact tests it should render correctly 1`] = `ShallowWrapper {}`;
